### ISSUE TITLE
Recursion error when running `griffe griffe`

### ISSUE THREAD

**Issue by WillDaSilva** [pawamoy/griffe#3]
To reproduce, install `griffe` version `0.8.1.dev16+g580251b` (`pip install .` on commit `580251b550f952d72bf550ea26a1c09660761164` - currently the head of the `master` branch), then run `griffe griffe`. This results in a recursion error.

The following is the error message, with the repeating section manually truncated:
```
INFO       Loading package griffe
Traceback (most recent call last):
  File "/home/will/.pyenv/versions/workshop/bin/griffe", line 8, in <module>
    sys.exit(main())
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/cli.py", line 234, in main
    packages = _load_packages(opts.packages, extensions, search, opts.docstyle, opts.docopts)
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/cli.py", line 97, in _load_packages
    if not loader.follow_aliases(obj, only_exported=False, only_known_modules=True):
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/loader.py", line 240, in follow_aliases
    member.resolve_target()  # type: ignore[union-attr]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 810, in resolve_target
    self._target = self.modules_collection[self._target_path]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/mixins.py", line 26, in __getitem__
    return self.members[parts[0]][parts[1]]  # type: ignore[attr-defined]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/mixins.py", line 26, in __getitem__
    return self.members[parts[0]][parts[1]]  # type: ignore[attr-defined]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 725, in __getitem__
    return self.target[key]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 790, in target
    self.resolve_target()
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 810, in resolve_target
[...]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 810, in resolve_target
    self._target = self.modules_collection[self._target_path]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/mixins.py", line 26, in __getitem__
    return self.members[parts[0]][parts[1]]  # type: ignore[attr-defined]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/mixins.py", line 26, in __getitem__
    return self.members[parts[0]][parts[1]]  # type: ignore[attr-defined]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 725, in __getitem__
    return self.target[key]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 790, in target
    self.resolve_target()
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 810, in resolve_target
    self._target = self.modules_collection[self._target_path]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/mixins.py", line 26, in __getitem__
    return self.members[parts[0]][parts[1]]  # type: ignore[attr-defined]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/mixins.py", line 26, in __getitem__
    return self.members[parts[0]][parts[1]]  # type: ignore[attr-defined]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 725, in __getitem__
    return self.target[key]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 790, in target
    self.resolve_target()
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 810, in resolve_target
    self._target = self.modules_collection[self._target_path]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/mixins.py", line 26, in __getitem__
    return self.members[parts[0]][parts[1]]  # type: ignore[attr-defined]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/mixins.py", line 26, in __getitem__
    return self.members[parts[0]][parts[1]]  # type: ignore[attr-defined]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 725, in __getitem__
    return self.target[key]
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 790, in target
    self.resolve_target()
  File "/home/will/.pyenv/versions/3.9.6/envs/workshop/lib/python3.9/site-packages/griffe/dataclasses.py", line 810, in resolve_target
    self._target = self.modules_collection[self._target_path]
RecursionError: maximum recursion depth exceeded
```

This error does not occur on version `0.8.0`, which has been published to PyPI, but instead a key error is raised on that version when running `griffe griffe`. Given that `0.8.0` is an older version, I won't create an issue for it unless you'd like me to.

**Comment by WillDaSilva** [comment]
Both `0.8.0` and `0.8.1.dev16+g580251b` work fine for some other packages, such as `six`.

**Comment by pawamoy** [comment]
It should be fixed using latest commits. Just tried in the gitpod workspace successfully.

**Comment by pawamoy** [comment]
Just tried this and it worked:

```console
$ cd griffe
$ python -m venv .venv
$ . .venv/bin/activate
$ pip install .
Successfully installed griffe-0.8.1.dev16+g8aa5ed0.d20220103
$ cd  # to avoid PDM loading __pypackages__
$ griffe griffe
# json output, no recursion
```

**Comment by pawamoy** [comment]
Should be fixed with griffe 0.9.0. Please reopen if you encounter the same issue again!
